components/style/servo/selector_parser.rs

```diff
@@ -100,44 +100,51 @@ impl PseudoElement {
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 #[allow(missing_docs)]
 pub enum NonTSPseudoClass {
-    AnyLink,
-    Link,
-    Visited,
     Active,
+    AnyLink,
+    Checked,
+    Disabled,
+    Enabled,
     Focus,
     Fullscreen,
     Hover,
-    Enabled,
-    Disabled,
-    Checked,
     Indeterminate,
-    ServoNonZeroBorder,
+    Lang(Atom),
+    Link,
+    PlaceholderShown,
     ReadWrite,
     ReadOnly,
-    PlaceholderShown,
+    ServoNonZeroBorder,
     Target,
+    Visited,
 }
 
 impl ToCss for NonTSPseudoClass {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
         use self::NonTSPseudoClass::*;
+        let lang_str = if let Lang(ref lang) = *self {
+            format!(":lang({})", lang)
```

Comment:
This seems somewhat flacky. also, I believe you need to escape the language (see `CSSStringWriter`. Verify that other browsers escape it and please add a test for this to WPT if there isn't any already.

Instead of this dummy variable, can't you do something like returning early in the `Lang` case?